microbootstrap/bootstrappers/base.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+import dataclasses
+import typing
+
+from microbootstrap.helpers import dataclass_to_dict_no_defaults, merge_dataclasses_configs, merge_dict_configs
+from microbootstrap.instruments import OpentelemetryInstrumentConfig, SentryInstrumentConfig
+
+
+if typing.TYPE_CHECKING:
+    from microbootstrap.instruments import Instrument
+    from microbootstrap.settings.base import BootstrapSettings
+
+
+ApplicationT = typing.TypeVar("ApplicationT")
+SettingsT = typing.TypeVar("SettingsT", bound="BootstrapSettings")
+SelfT = typing.TypeVar("SelfT", bound="ApplicationBootstrapper[typing.Any, typing.Any, typing.Any]")
+
+
+@dataclasses.dataclass()
+class ApplicationBootstrapper(typing.Protocol[SettingsT, ApplicationT, dataclasses._DataclassT]):
+    settings: SettingsT
+    application_type: type[ApplicationT] = dataclasses.field(init=False)
+    application_config: dataclasses._DataclassT = dataclasses.field(init=False)
+
+    sentry_instrument_type: type[Instrument[SentryInstrumentConfig]] = dataclasses.field(init=False)
```

Comment:
I actually don't like these *_type variables because in the near future there will be 5-6 of these tool types.
I wonder how we can make this part of the code cleaner
Any ideas?

Reply:
1. Make instruments return from `bootstrap()` something generic instead of pieces of application config. 
2. Add private methods like this: `def _bootstrap_sentry(self) -> ApplicationConfigPart | None: …` for each of instruments to base application bootstrapper.

This way we won’t have to add classes for every framework+instrument (like `LitestarSentryInstrument`).

Reply:
Adding to the second point. In base bootstrapper methods should be abstract

Reply:
Want to add new framework? All you need is to write one class.
Want to add new instrument? Write a method for each framework. Or just add placeholder that returns none

Reply:
Separation of concerns, baby ;)

Reply:
Answering the initial comment:
1) It's already working like this, cause those methods are returning some dict[str, Any]
2) In your approach, you have to add another method instead of one type, hence more code is emerging

About abstract methods, yeah, also might work, but i was thinking, that bootstrapper is some kind of a protocol that you can apply to any framework

Now if you want to add something, you just have to make a bootstrapper class and make instruments return things that are compatible with application input arguments

So, it's already working as you described, but with types instead of methods

But still, i don't like the concept of types/extra _bootstrap_instrument methods
Maybe, it would be nice to register instruments via some decorated, idk

Reply:
> About abstract methods, yeah, also might work, but i was thinking, that bootstrapper is some kind of a protocol that you can apply to any framework

Yes!

I know it looks uglier the way I describe it. Though, it seems to be a better abstraction from my perspective.

Reply:
> But still, i don't like the concept of types/extra _bootstrap_instrument methods
Maybe, it would be nice to register instruments via some decorator, idk

Where would you put the framework-specific code then?
In my suggestion, methods are used exactly for registration. It’s opt-in. One potential problem is, that specific set of instruments is assumed, i. e. users won’t be able to extend this set.

Let’s discuss this particular issue at our call

Reply:
It may be that I didn’t explain well the picture I see

Reply:
Okay, i'll stop adding instruments for now :)